cmd/stack_package.go

```diff
@@ -399,9 +429,121 @@ func getLabelsForStackImage(stackID string, buildImage string, stackYaml StackYa
 	configLabels[appsodyStackKeyPrefix+"tag"] = buildImage
 
 	for key, value := range configLabels {
-		labelString := fmt.Sprintf("%s=%s", key, value)
-		labels = append(labels, labelString)
+		labels[key] = value
 	}
 
 	return labels, nil
 }
+
+// createTemplateMap uses the git labels, stack.yaml, stackID and imageNamespace to create a map
+// with all the necessary data needed for the template
+func createTemplateMap(labels map[string]string, stackYaml StackYaml, imageNamespace string) map[string]interface{} {
+
+	// split version number into major, minor and patch strings
+
+	versionLabel := labels[ociKeyPrefix+"version"]
+	versionFull := strings.Split(versionLabel, ".")
+
+	// Create map that holds stack variables
+
+	// create stack variables and add to templateMetadata map
+	var templateMetadata = make(map[string]interface{})
+
+	var stack = make(map[string]interface{})
+	stack["id"] = labels[appsodyStackKeyPrefix+"id"]
+	stack["name"] = labels[ociKeyPrefix+"title"]
+	stack["version"] = versionLabel
+	stack["description"] = labels[ociKeyPrefix+"description"]
+	stack["created"] = labels[ociKeyPrefix+"created"]
+	stack["tag"] = labels[appsodyStackKeyPrefix+"tag"]
+	stack["maintainers"] = labels[ociKeyPrefix+"authors"]
+	// create version map and add to templateMetadata map
+	var semver = make(map[string]string)
+	semver["major"] = versionFull[0]
+	semver["minor"] = versionFull[1]
+	semver["patch"] = versionFull[2]
+	semver["majorminor"] = strings.Join(versionFull[0:2], ".")
+	semver["full"] = versionLabel
+	stack["semver"] = semver
+	// create image map add to templateMetadata map
+	var image = make(map[string]string)
+	image["namespace"] = imageNamespace
+	stack["image"] = image
+
+	// loop through user variables and add them to map, must begin with alphanumeric character
+	for key, value := range stackYaml.TemplatingData {
+
+		// validates that key starts with alphanumeric character
+		runes := []rune(key)
+		firstRune := runes[0]
+		if unicode.IsLetter(firstRune) || unicode.IsNumber(firstRune) {
+			stack[key] = value
+		}
+	}
+
+	templateMetadata["stack"] = stack
+	return templateMetadata
+
+}
+
+// applyTemplating walks through the copied folder directory and applies a template using the
+// previously created templateMetada to all files in the target directory
+func applyTemplating(projectPath string, templateMetadata interface{}) error {
+
+	// sets stack path to be the copied folder
+	stackPath := projectPath + "stackPath"
```

Comment:
Good job getting rid of the `projectPath + "stackPath"` everywhere else, but you are still left with this one. I think it would be best to pass this in as a func parameter. The point is, if we ever change the location, it only has to be changed once in the code.